syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "custom_types.proto";

service DepositService {
  rpc GetAll (GetAllRequest) returns (GetAllDepositsResponse);
  rpc GetById (GetDepositByIdRequest) returns (DepositModel);
  rpc GetByCustomer (GetDepositsByCustomerRequest) returns (GetAllDepositsResponse);
}

message DepositModel {
  int32 deposit_id = 1;
  optional int32 customer_id = 2;
  optional string amount = 3;
  optional string interest_rate = 4;
  optional DateOnly start_date = 5;
  optional DateOnly end_date = 6;
  optional string status = 7;
}

message GetAllDepositsResponse {
  repeated DepositModel deposits = 1;
}

message GetDepositByIdRequest {
  int32 deposit_id = 1;
}

message GetDepositsByCustomerRequest {
  int32 customer_id = 1;
}
